### PR TITLE
building docker image to run react app on cntainer

### PR DESCRIPTION
**Pull request by hossainahmed18**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [ ] Refactor
- [x] Infrastructure
- [ ] Other (please specify)

## What was done

Dockerize the previously build todo-app of react. A Dockerfile created and a dockerignore. necessary changes are made to build a docker image and run it on docker container

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: junayed-task/todo-docker/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.gitignore
+src/App.test.js
```
Comment (musfique-m) [inline junayed-task/todo-docker/.dockerignore:2]: There is no need add test files to dockerignore.
Later we may decide to run the test from our image. We won't be able to do that in this case.
Please remove the test file from here.

File: junayed-task/todo-docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#pulling the base image from docker hub
+FROM node:alpine
+
+
+#creating a work directory in docker container
+WORKDIR /app
+
+
+#This will prevent unncessarily runing "npm install" if there is no change in pachage.json file
+COPY package.json .
```
Comment (musfique-m) [inline junayed-task/todo-docker/Dockerfile:10]: Please also copy `package-lock.json`. Its goal is to keep track of exact version of the package installed.
[Further information](https://nodejs.dev/en/learn/the-package-lock-json-file/)

**Review by musfique-m — CHANGES_REQUESTED** [review]

File: junayed-task/todo-docker/.dockerignore (hunk 1 above)
Comment (hossainahmed18) [reply junayed-task/todo-docker/.dockerignore]: resolved

File: junayed-task/todo-docker/Dockerfile (hunk 2 above)
Comment (hossainahmed18) [reply junayed-task/todo-docker/Dockerfile]: resolved

**Review by musfique-m — APPROVED** [review]
Looks good. Approved by code inspection.

**Pull request merged**